packages/solana/src/__test__/deposit.test.ts

```diff
@@ -0,0 +1,476 @@
+import {
+  type ConfirmedSignatureInfo,
+  Connection,
+  MessageV0,
+  PublicKey,
+  type VersionedTransactionResponse,
+} from '@solana/web3.js';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { findSolanaDepositSignature } from '../deposit';
+
+vi.mock('@solana/web3.js', async (importOriginal) => {
+  const actual = await importOriginal();
+
+  const ConnectionMock = vi.fn();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  ConnectionMock.prototype.getSignaturesForAddress = vi.fn();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  ConnectionMock.prototype.getTransactions = vi.fn();
+
+  // @ts-expect-error typescript cannot infer the type of actual
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  return { ...actual, Connection: ConnectionMock };
+});
+
+const transactions: Record<string, VersionedTransactionResponse> = {
+  '3zC67SpVaUAMXitSZw6Z1HtTPFtEEeKdJ2cAkHGAPYyBdWkbbT3RRva9Do6SbQjk8Zs8oHJrprV8txdc3YgBp33x': {
+    blockTime: 1733324203,
+    meta: {
+      computeUnitsConsumed: 450,
+      err: null,
+      fee: 45000,
+      innerInstructions: null,
+      loadedAddresses: {
+        readonly: [],
+        writable: [],
+      },
+      logMessages: null,
+      postBalances: [13136009466, 6000000000, 1, 1],
+      postTokenBalances: [],
+      preBalances: [19136054466, 0, 1, 1],
+      preTokenBalances: [],
+    },
+    slot: 305374576,
+    transaction: {
+      message: new MessageV0({
+        header: {
+          numReadonlySignedAccounts: 0,
+          numReadonlyUnsignedAccounts: 2,
+          numRequiredSignatures: 1,
+        },
+        staticAccountKeys: [
+          new PublicKey('HGDmbWtWotT3YH4SMinHsJ1QjziYtpHPHxBiy63tGFpn'),
+          new PublicKey('8z1bM4fWJwvz6shKBvdfFBtW1mNAvhZ6dw8TJo7ZxP4R'),
+          new PublicKey('11111111111111111111111111111111'),
+          new PublicKey('ComputeBudget111111111111111111111111111111'),
+        ],
+        recentBlockhash: 'HPn7zUWcbtNDGMUJoH9t6m7c33LA6MPmnRWCiZErHhLG',
+        compiledInstructions: [
+          {
+            accountKeyIndexes: [],
+            data: Uint8Array.from('3QAwFKa3MJAs'),
+            programIdIndex: 3,
+          },
+          {
+            accountKeyIndexes: [],
+            data: Uint8Array.from('Fj2Eoy'),
+            programIdIndex: 3,
+          },
+          {
+            accountKeyIndexes: [0, 1],
+            data: Uint8Array.from('3Bxs3zyoqgkX7Cqd'),
+            programIdIndex: 2,
+          },
+        ],
+        addressTableLookups: [],
+      }),
+      signatures: [
+        '3zC67SpVaUAMXitSZw6Z1HtTPFtEEeKdJ2cAkHGAPYyBdWkbbT3RRva9Do6SbQjk8Zs8oHJrprV8txdc3YgBp33x',
+      ],
+    },
+    version: 'legacy',
+  },
+  '': {
```

Comment:
probably not needed?

Reply:
good catch! its not strictly needed but we want to return it from one of the `getTransactions` calls to have a realistic spam test